domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -198,19 +218,23 @@ constructor(
                     cameraProvider.availableCameraInfos
                 ).first()
 
+                // get device-supported fixed frame rates
                 settingsRepository.updateSupportedFixedFrameRate(
-                    getSupportedFrameRates(),
+                    getDeviceSupportedFrameRates(),
                     sessionSettings.targetFrameRate
                 )
+
+                // get device-supported stabilization modes
+                val supportedStabilizationModes = getDeviceSupportedStabilizations()
+
                 settingsRepository.updatePreviewStabilizationSupported(
-                    isPreviewStabilizationSupported(cameraInfo)
+                    supportedStabilizationModes.contains(SupportedStabilizationMode.ON)
                 )
                 settingsRepository.updateVideoStabilizationSupported(
-                    isVideoStabilizationSupported(cameraInfo)
+                    supportedStabilizationModes.contains(SupportedStabilizationMode.ON)
```

Comment:
This should be:
`supportedStabilizationModes.contains(SupportedStabilizationMode.HIGH_QUALITY)`, correct?